aws/src/test/java/org/apache/iceberg/aws/glue/GlueCatalogTest.java

```diff
@@ -316,8 +347,11 @@ public void dropNamespace() {
   public void dropNamespace_notEmpty() {
     Mockito.doReturn(GetTablesResponse.builder()
         .tableList(
-            Table.builder().databaseName("db1").name("t1").build(),
-            Table.builder().databaseName("db1").name("t2").build()
+            Table.builder().databaseName("db1").name("t1").parameters(
+                ImmutableMap.of(
+                    BaseMetastoreTableOperations.TABLE_TYPE_PROP, BaseMetastoreTableOperations.ICEBERG_TABLE_TYPE_VALUE
```

Comment:
I think we should also test the case where the namespace is non-empty, but there is no Iceberg table.

Reply:
Great catch, dropping namespace with non-empty non-Iceberg table is interesting, in theory Iceberg should not care about it, but in practice we should probably not delete the namespace. So I updated the logic to check for iceberg or non-iceberg tables, and throw exception with different message for different cases. I have updated both the unit test and also the integration test.